_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -0,0 +1,435 @@
+			<a href="http://facebook.com" aria-label="Follow us">
+				<circle cx="50" cy="40" r="35" />
+			</a>
+		</svg>
+	</p>
+</html>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] but go to different resources.
+
+```html
+<html lang="en">
+	<p>
+		We are on social media:
+		<a href="http://facebook.com">Follow us</a>
+		<a href="http://twitter.com">Follow us</a>
+	</p>
+</html>
+```
+
+#### Failed Example 2
+
+The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They link to web pages that are similar, but have different information in their content.
+
+```html
+<html lang="en">
+	<p>
+		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html"
+			>Contact us</a
+		>) and get in touch (
+		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
+			>Contact us</a
+		>)
+	</p>
+</html>
+```
+
+#### Failed Example 3
+
+The two HTML `span` elements have an [explicit role][] of link, same [accessible name][] and [context][programmatically determined link context], but lead to resources that offer different content.
+
+```html
+<html lang="en">
+	<p>
+		<span
+			role="link"
+			tabindex="0"
+			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html'"
+		>
+			My university
+		</span>
+
+		<span
+			role="link"
+			tabindex="0"
+			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html'"
+		>
+			My university
+		</span>
+	</p>
+</html>
+```
+
+#### Failed Example 4
+
+The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.
+
+```html
+<html lang="en">
+	<p>
+		We are on social media:
+		<a href="http://facebook.com"><img src="facebook.jpg" alt="Follow us"/></a>
+		<a href="http://twitter.com"><img src="twitter.jpg" alt="Follow us"/></a>
+	</p>
+</html>
+```
+
+#### Failed Example 5
+
+The two SVG `a` elements have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.
+
+```html
+<html lang="en">
+	<p>
+		<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+			<a href="http://facebook.com" aria-label="Follow us">
+				<circle cx="50" cy="40" r="35" />
+			</a>
+
+			<a href="http://twitter.com">
+				<text x="50" y="90" text-anchor="middle">
+					Follow us
+				</text>
+			</a>
+		</svg>
+	</p>
+</html>
+```
+
+#### Failed Example 6
+
+The two HTML `a` elements with the same [accessible name][] and [context][programmatically determined link context] resolve to the [same resource][] after redirect, but the redirect is not instant.
+
```

Comment:
isn't what counts the end result? not sure i agree with this as a failure

Reply:
But we can't tell when the redirect will happen if it does not happen instantly. That's why we have a note stating "Resolving the links includes potential redirects, if the redirects happen instantly."

Imagine a redirect that happens half an hour after the link has been followed. Almost all users will not even perceive that redirect, so they will experience different content and the SC should fail.

(of course we can tell when the redirect happens, but then the issue becomes where do we draw the line between redirects that will be perceived and those that won't)